src/pv/pvData.h

```diff
@@ -32,6 +32,12 @@
 typedef class std::ios std::ios_base;
 #endif
 
+#if defined(_WIN32) && !defined(_MINGW)
+#pragma warning(disable: 4344) // A call to a function using explicit template
+// arguments calls a different function than it would if explicit arguments had not been specified
+// Removes warnings for template getSubField.
```

Comment:
Actually this removes the warning for all source code after this point in the file being compiled, and since this is a header file it does that for any source file that includes this header. I'm not saying this is _necessarily_ a bad thing as I agree that it appears to be a rather stupid warning, but disabling it this way may not be the best solution. Normally we turn off MSVC warnings for a section of code with
```cpp
#if defined(_MSC_VER)
#   pragma warning(push)
#   pragma warning(disable: 4344)
#endif
```
followed after the code in question by
```cpp
#if defined(_MSC_VER)
#   pragma warning(pop)
#endif
```
but unfortunately in this case I believe we would have to wrap every location where that the template is _called_ rather than wrapping the template _definition_ in the appropriate header file(s), and that isn't really feasible.

If we're going to disable the warning globally I would rather it be done with a `/w44344` flag on the compiler command-line. I haven't worked out what the Makefile change would need to be, because it would probably have to be applied to all modules, or be done in Base (where we already have the flags `/W3 /w44355` — see [this MSDN page](https://msdn.microsoft.com/en-us/library/thxezb7y.aspx) for a description of how these flags work).

This may be more of a philosophical point of view though, so I'd appreciate opinions from @ralphlange , @msekoranja and/or @mdavidsaver .